app/controllers/application_controller.rb

```diff
@@ -5,6 +5,133 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
+  end
+get '/tweets/new' do 
+    if Helpers.is_logged_in?(session)
+      erb :'/tweets/create_tweet'
+    else 
+      redirect '/login'
+    end
+  end
+
+  post '/tweets' do
+    if params[:content] == ""
+      redirect :'/tweets/new'
+    else
+      user = User.find_by_id(session[:user_id])
+      Tweet.create(content: params[:content], user_id: user.id)
+    end
+
+  end
+
+  get '/tweets' do 
+
+    if session[:user_id]
+      @tweets = Tweet.all
+      erb :"tweets/tweets"
+    else
+      redirect '/login'
+    end
   end
 
+  get '/users/:username' do
+    @user = User.find_by(:username => params[:username])
+    erb :'/users/show'
+  end
+
+  get '/tweets/:id' do
+    if Helpers.is_logged_in?(session)
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/show_tweet'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do 
+    if Helpers.is_logged_in?(session) 
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
+        erb :'/tweets/edit_tweet'
+      else
+        redirect to '/tweets'
+      end
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets/:id' do
+    if params[:content] != ""
+      @tweet = Tweet.find_by_id(params[:id])
+      @tweet.content = params[:content]
+      @tweet.save
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a POST request than going through an HTML form...